lib/db/db.rb

```diff
@@ -18,11 +18,11 @@ def at(index)
   end
 
   def add(contact)
-    writer.run(added_a(contact)) if !writer.nil?
+    writer.write(added_a(contact)) if !writer.nil?
   end
 
   def update(contact)
-    writer.run(updated_with(add_id_to(contact))) if !writer.nil?
+    writer.write(updated_with(add_id_to(contact))) if !writer.nil?
```

Comment:
can we improve the design of the code to avoid having to do these nil checks? a db without a reader and writer shouldn't be able to exist; if that's the case, nil checks are superfluous.